pkg/apis/azure/types_cloudprofile.go

```diff
@@ -31,6 +31,8 @@ type CloudProfileConfig struct {
 	// MachineImages is the list of machine images that are understood by the controller. It maps
 	// logical names and versions to provider-specific identifiers.
 	MachineImages []MachineImages
+	// AcceleratedNetworkingMachineTypes is a list of machine types which support accelerated networking.
+	AcceleratedNetworkingMachineTypes []string
 }
```

Comment:
To allow further enhancements it's better to use here: 

```suggestion
	// AcceleratedNetworkingMachineTypes is a list of machine types which support accelerated networking.
	AcceleratedNetworkingMachineTypes []AcceleratedNetworkingMachineType
}

// doc string
type AcceleratedNetworkingMachineType struct {
  Name string
}
```

Reply:
Also, can you implement a validation that only machine types that are also configured in `.spec.machineTypes` are present in this list? Does something like this make sense?

Reply:
Hmm yes, we could do that, but this is basically a list of machine types which support accelerated networking and not the list of usable machine types. These information could be fetched automatically for all machine types e.g. via `az vm list-skus`. The list of usable machine types is most probably maintained manually and only selected machine types are part of it.

I'm also ok with having that validation.

Reply:
See https://github.com/gardener/gardener-extension-provider-azure/pull/65#discussion_r408030329 - your decision :)